From: Dan Carpenter <error27@gmail.com>
Subject: [PATCH 061/111]     9p: potential ERR_PTR() dereference
Git-commit: 8f587df479c3cea14ba1a9b9d58f34fd2fd6d58b
Patch-Mainline: 2.6.36
References: FATE#311639

    p9_client_walk() can return error values if we run out of space or there
    is a problem with the network.

	upstream commit 8f587df479c3cea14ba1a9b9d58f34fd2fd6d58b
    Signed-off-by: Dan Carpenter <error27@gmail.com>
    Signed-off-by: Eric Van Hensbergen <ericvh@gmail.com>
Signed-off-by: Harsh Prateek Bora <harsh@linux.vnet.ibm.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 fs/9p/fid.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/fs/9p/fid.c b/fs/9p/fid.c
index abc2cca..1b1e3df 100644
--- a/fs/9p/fid.c
+++ b/fs/9p/fid.c
@@ -241,7 +241,8 @@ struct p9_fid *v9fs_fid_lookup(struct dentry *dentry)
 	}
 	kfree(wnames);
 fid_out:
-	v9fs_fid_add(dentry, fid);
+	if (!IS_ERR(fid))
+		v9fs_fid_add(dentry, fid);
 err_out:
 	up_read(&v9ses->rename_sem);
 	return fid;
-- 
1.7.1.1

